program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,155 @@
+"""
+Tests for the Program-Intent API views
+"""
+import datetime
+import uuid
+
+import ddt
+from django.urls import reverse
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from test_utils import ProgramIntentAPITestCase
+from test_utils.factories import UserFactory
+
+
+@ddt.ddt
+class ProgramIntentRecentAndCertainViewTests(ProgramIntentAPITestCase):
+    """
+    Tests MostRecentAndCertainIntentsView
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.program_id1 = str(uuid.uuid4())
+        self.program_id2 = str(uuid.uuid4())
+
+        self.p1_intent_maybe = ProgramIntent.objects.create(
+            lms_user_id=self.user.lms_user_id,
+            program_uuid=self.program_id1,
+            reason="a reason",
+            certainty="MAYBE",
+            effective_timestamp=datetime.datetime(2020, 8, 17),
+        )
+
+    def get_api(self, user):
+        """
+        Helper function to make a get request to the API
+        """
+        url = reverse("api:v1:intents-list")
+
+        headers = self.build_jwt_headers(user)
+        return self.client.get(url, **headers)
+
+    def get_response(self, user, expected_response):
+        """
+        Helper function to get API response
+        """
+
+        response = self.get_api(user)
+        self.assertEqual(response.status_code, expected_response)
```

Comment:
Nit: It's not a huge deal, but having an assertion in a function called `get_response` is unexpected to me. I think it would be more clear what a test is actually testing if the assertion is defined within the test. That way, it's more clear in a given test that the test is expecting a `200` response, for example. Another way of thinking about it is, if you read the line `self.get_response(self.user, 200)` without necessarily knowing was `get_response` does, would that make sense to you? I don't think it's a big deal, but it's just something to think about. Let me know what you think!

Reply:
this is a good point and also made me realize that now with Andy's earlier suggestion the way i have my grab_best_intents method set up i can test it directly without a request in some of these tests!

Reply:
are you suggesting i remove the get_response method entirely?

Reply:
I'm suggesting that you remove the line `self.assertEqual(response.status_code, expected_response)` and just make that assertion directly in your test. That way, it's clear in the actual test what you are testing. I think it would be clearer to collocate all assertions your test needs to make within the test.

`self.get_response(self.user, 200)` doesn't really tell me the purpose of the `200`. It's not clear what the purpose of the `200` is in a method called `get_response`.`get_response`'s obvious responsibility is to make a request and return the response. If it also starts asserting things about the response, it's doing more than the name claims that it does, and it becomes harder to read, in my opinion.

Reply:
Thank you, this makes sense! Just added a commit 👍